mParticle-FollowAnalytics/MPKitFollowAnalytics.m

```diff
@@ -0,0 +1,141 @@
+    MPKitExecStatus *execStatus = nil;
+
+    Class klass = NSClassFromString(@"FollowAnalytics");
+    if (!klass) {
+        NSLog(@"You must initialize the FollowAnalytics SDK before starting mParticle. Please refer to https://dev.followanalytics.com/sdks/ios/documentation.");
+        execStatus = [[MPKitExecStatus alloc] initWithSDKCode:[[self class] kitCode] returnCode:MPKitReturnCodeRequirementsNotMet];
+        return execStatus;
+    }
+
+    [self start];
+
+    execStatus = [[MPKitExecStatus alloc] initWithSDKCode:[[self class] kitCode] returnCode:MPKitReturnCodeSuccess];
+    return execStatus;
+}
+
+
+- (id const)providerKitInstance {
+    return nil;
+}
+
+
+#pragma mark Application
+#pragma mark User attributes and identities
+ - (MPKitExecStatus *)setUserAttribute:(NSString *)key value:(NSString *)value {
+     [FollowAnalytics.userAttributes setString:value
+                                        forKey:key];
+
+     MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceFollowAnalytics) returnCode:MPKitReturnCodeSuccess];
+     return execStatus;
+ }
+
+ - (MPKitExecStatus *)removeUserAttribute:(NSString *)key {
+     [FollowAnalytics.userAttributes clear:key];
+
+     MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceFollowAnalytics) returnCode:MPKitReturnCodeSuccess];
+     return execStatus;
+ }
+
+ - (MPKitExecStatus *)setUserIdentity:(NSString *)identityString identityType:(MPUserIdentity)identityType {
+     switch (identityType) {
+         case MPUserIdentityCustomerId:
+             [FollowAnalytics setUserId:identityString];
+             break;
+         default:
+             break;
+     }
+
+      MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceFollowAnalytics) returnCode:MPKitReturnCodeSuccess];
+      return execStatus;
+ }
+
+#pragma mark e-Commerce
+#pragma mark Events
+- (MPKitExecStatus *)logEvent:(MPEvent *)event {
+    [FollowAnalytics logEvent:event.name
+                      details:event.info];
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceFollowAnalytics) returnCode:MPKitReturnCodeSuccess];
+    return execStatus;
+}
+
+- (MPKitExecStatus *)logError:(NSString *)message eventInfo:(NSDictionary *)eventInfo {
+    [FollowAnalytics logError:message details:eventInfo];
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceFollowAnalytics) returnCode:MPKitReturnCodeSuccess];
+    return execStatus;
+}
+
+- (MPKitExecStatus *)logException:(NSException *)exception {
+    [FollowAnalytics logError:[exception name] details:[exception reason]];
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceFollowAnalytics) returnCode:MPKitReturnCodeSuccess];
+    return execStatus;
+}
+
+- (MPKitExecStatus *)logScreen:(MPEvent *)event {
```

Comment:
You don't need to implement methods if you don't have any functionality to execute in them, we will check responds to selector before calling.